dev_docs/contributing/kibana_http_api_guidelines.mdx

```diff
@@ -0,0 +1,502 @@
+---
+id: kibHttpApiGuidelines
+slug: /kibana-dev-docs/contributing/http-api-guidelines
+title: Guidelines for Terraform friendly HTTP APIs
+description: Guidelines for designing Terraform friendly HTTP APIs
+date: 2025-06-16
+tags: ['kibana','contributor', 'dev', 'http', 'api']
+---
+
+Kibana's role has expanded beyond the UI. APIs have become the backbone for how customers automate their workflows, build custom scripts, and create their own integrations. With GitOps becoming the standard and Terraform establishing itself as the go-to infrastructure-as-code tool, HTTP APIs are now business-critical. The rise in Search AI means that it's more important than ever to make sure our APIs are easily understood by LLMs.
+
+For Terraform to be a first class citizen, the provider needs to cover resources that weren't available before—think Dashboards, Visualizations, Data Views, and Detection Rules. This means we need to build REST APIs that follow the OpenAPI Specification properly, making it straightforward for teams to manage these resources as code through GitOps workflows.
+
+These guidelines will help you build APIs that are easy to use, whether someone's automating with scripts or managing infrastructure at scale.
+
+## Terraform Provider Developer-Friendly API Design
+
+Terraform can work with any API, but some APIs are easier to deal with than others. Here are some tips to make your API more Terraform provider-friendly:
+
+### Think in terms of resources
+
+APIs that stick to a consistent set of arguments or parameters are much easier to map into Terraform. Terraform can describe the "thing" your API is managing. RPC-based APIs are harder to work with since they’re not declarative. Your HTTP APIs should describe REST-like actions (GET, POST, DELETE, etc.) against resources not remote procedures like: executeJob. 
+
+APIs designed around resources are easier to support than those focused on action-oriented endpoints.
+
+✅ Preferred: REST-like actions against resources
+
+```
+GET    /api/alerting/rule/{id}     # Kibana Alerting API
+POST   /api/alerting/rule/{id}     # Create rule
+PUT    /api/alerting/rule/{id}     # Update rule
+DELETE /api/alerting/rule/{id}     # Delete rule
+```
+
+⚠️ Alternative: action-style endpoints
+
+```
+POST /api/executeJob
+POST /api/processSpaceUpdate
+POST /api/triggerIndexRebalance
+```
+
+Mapping resource-oriented APIs to a declarative model means users can describe the desired state rather than the steps to achieve it. While action-oriented endpoints sometimes become necessary, knowing the trade-offs helps you make informed decisions.
+
+#### Ideal
+
+The [Index Lifecycle Management API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-ilm-put-lifecycle) (PUT /\_ilm/policy/{policy}) declaratively defines the desired lifecycle state rather than imperatively executing phase transitions:
```

Comment:
```suggestion
The [Index Lifecycle Management API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-ilm-put-lifecycle) (PUT /\_ilm/policy/\{policy\}) declaratively defines the desired lifecycle state rather than imperatively executing phase transitions:
```